### PR TITLE
feat: produce and serve `did:web` DID documents

### LINKED ISSUE #18: Add support for `did:web`
### Description

Add support for the `did:web` method. The agent should be able to create, destroy and verify identities using the `did:web` method, from the perspective of all 3 roles: Issuer, Holder and Verifier. Once created, the identity must be stored and be re-instantiated if the agent is turned off and on.

### Motivation

One step closer to full compatibility with the Dutch Blockchain Coalition's DIIP profile.

### Resources

- [DIIP Profile](https://github.com/DutchBlockchainCoalition/DIIP/discussions/12)
- [DID method](https://w3c-ccg.github.io/did-method-web/)

### To-do List

- [x] Find existing `did:web` crate in Rust (or don't and implement the method separately)
- [x] Implement `did:web` as another method option
- [x] Add verification of `did:web` method

### PR DESCRIPTION
**Pull request by daniel-mader**
# Description of change
Adds support for the `did:web` method by supplying the DID document on the expected endpoint. Also resolves the document for a given DID.

## Links to any relevant issues
- #18

### Blocked by
- [x] https://github.com/impierce/did-manager/pull/20

## How the change has been tested
n/a

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: agent_api_rest/postman/ssi-agent.postman_collection.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
 	"info": {
-		"_postman_id": "53b46e18-de7f-4973-8304-8238844a71ce",
+		"_postman_id": "a196fe6f-81d0-4e6f-b57d-78d557a98789",
 		"name": "ssi-agent",
-		"schema": "https://schema.getpostman.com/json/collection/v2.0.0/collection.json",
-		"_exporter_id": "24972330"
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
```
Comment (nanderstabel) [inline agent_api_rest/postman/ssi-agent.postman_collection.json:5]: `[resolved]`
Just curious, what is the significance of using v2.1?

File: agent_secret_manager/src/subject.rs [hunk 2]
```diff
@@ -50,8 +51,21 @@ impl Sign for Subject {
...
+                .produce_document(
+                    method,
+                    Some(did_manager::MethodSpecificParameters::Web { origin: origin() }),
+                )
+                .await
+                .ok()
+                .and_then(|document| document.verification_method().first().cloned())
+                .map(|first| first.id().to_string());
+        }
+
```
Comment (nanderstabel) [inline agent_secret_manager/src/subject.rs:66]: `[resolved]` 

In order to counter some repeated code you could consider implementing this method like so:
```rust
    async fn identifier(&self, subject_syntax_type: &str) -> anyhow::Result<String> {
        let method: DidMethod = serde_json::from_str(&format!("{subject_syntax_type:?}"))?;

        let method_specific_parameters =
            (method == DidMethod::Web).then(|| did_manager::MethodSpecificParameters::Web { origin: origin() });

        Ok(self
            .secret_manager
            .produce_document(method, method_specific_parameters)
            .await
            .map(|document| document.id().to_string())?)
    }
```

where
```rust
        let method_specific_parameters =
            (method == DidMethod::Web).then(|| did_manager::MethodSpecificParameters::Web { origin: origin() });
```

will result in `None` for all DID Methods other than did:web

**Review by nanderstabel — DISMISSED** [review]

File: agent_api_rest/postman/ssi-agent.postman_collection.json (hunk 1 above)
Comment (daniel-mader) [reply agent_api_rest/postman/ssi-agent.postman_collection.json:5]: Nothing particular, I just upgraded to the latest stable.

**Review by nanderstabel — DISMISSED** [review]

**Review by nanderstabel — APPROVED** [review]

File: agent_secret_manager/src/subject.rs (hunk 2 above)
Comment (daniel-mader) [reply agent_secret_manager/src/subject.rs:66]: Thanks! Will consider it in the follow-up PR 👍

**Pull request merged**